### ISSUE TITLE
Combine .filter() and .map() to create atomic selective DELETEs

### ISSUE THREAD

**Issue by Pyknic** [speedment/speedment#183]
In many cases you would want to delete only a very specific row in the database. Take the following code for an example:

``` sql
DELETE * FROM `user` WHERE `id` = 5 AND `avatar` IS NULL;
```

We want to delete the user with id "5" only if the avatar is set to null. If we do this in Speedment right now, it would look like this:

``` java
users.stream()
    .filter(User.ID.equal(5))
    .filter(User.AVATAR.isNull())
    .forEach(users::delete);
```

The problem is that the generated code will do this in two steps. First, it will download every user that matches the condition and then it will remove those users. This is equivalent to the following:

``` sql
SELECT * FROM `user` WHERE `id` = 5 AND `avatar` IS NULL;
// Some time passes
DELETE * FROM `user` WHERE `id` = 5;
```

Here we have a problem. If another client updates the user row so that avatar is no longer null, the user might still be deleted since there will be a race condition. It would be better if the manager could translate the complete stream into one SQL statement and execute it atomically.

**Comment by minborg** [comment]
Will be fixed by #539

**Comment by minborg** [comment]
I am closing this now because #539 is fixed
